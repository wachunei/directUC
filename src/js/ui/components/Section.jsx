import styled from "styled-components";

const Section = styled.div`
  padding: 15px 20px;
  background-color: ${({ theme, ghost }) => !ghost && theme.sectionBackground};
  margin-bottom: 15px;
`;

export default Section;
