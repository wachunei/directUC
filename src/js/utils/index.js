import browser from "webextension-polyfill";

export const removeSuffix = (suffix, value) =>
  value.endsWith(suffix) ? value.slice(0, -suffix.length) : value;

export const parseDistinguishedName = (name) =>
  name.replace(
    /\w\S*/g,
    (txt) => txt.charAt(0).toUpperCase() + txt.substr(1).toLowerCase()
  );

export const isFirefox = process.env.TARGET === "firefox";

export const trackingId =
  process.env.NODE_ENV === "development" ? "UA-177532719-1" : "UA-62971405-1";

export const { version } = browser.runtime.getManifest();
