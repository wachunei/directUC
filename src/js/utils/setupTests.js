// eslint-disable-next-line import/no-extraneous-dependencies
import "jest-styled-components";
