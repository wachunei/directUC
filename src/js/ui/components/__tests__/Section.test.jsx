import React from "react";
import Section from "../Section";
import { renderTree } from "./utils";
import "jest-styled-components";

describe("Section Component", () => {
  test("should match snapshot", () => {
    const tree = renderTree(<Section />);
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with ghost", () => {
    const tree = renderTree(<Section ghost />);
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with lightTheme", () => {
    const tree = renderTree(<Section />, "lightTheme");
    expect(tree).toMatchSnapshot();
  });
  test("should match snapshot with darkTheme", () => {
    const tree = renderTree(<Section />, "darkTheme");
    expect(tree).toMatchSnapshot();
  });
});
