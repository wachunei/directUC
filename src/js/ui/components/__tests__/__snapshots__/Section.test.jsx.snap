// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Component should match snapshot 1`] = `
.c0 {
  padding: 15px 20px;
  background-color: #FAFAFA;
  margin-bottom: 15px;
}

<div
  className="c0"
/>
`;

exports[`Section Component should match snapshot with darkTheme 1`] = `
.c0 {
  padding: 15px 20px;
  background-color: #202020;
  margin-bottom: 15px;
}

<div
  className="c0"
/>
`;

exports[`Section Component should match snapshot with ghost 1`] = `
.c0 {
  padding: 15px 20px;
  margin-bottom: 15px;
}

<div
  className="c0"
/>
`;

exports[`Section Component should match snapshot with lightTheme 1`] = `
.c0 {
  padding: 15px 20px;
  background-color: #FAFAFA;
  margin-bottom: 15px;
}

<div
  className="c0"
/>
`;
